### PR TITLE
GetPageRangesDiff and GetManagedDiskRangesDiff

### PR DESCRIPTION
**Pull request by Jinming-Hu**
closes https://github.com/Azure/azure-sdk-for-cpp/issues/1030

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp [hunk 1]
```diff
@@ -1380,6 +1408,34 @@ namespace Azure { namespace Storage { namespace Blobs {
      */
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:1408]: This should be moved to the API's parameter list and make it mandatory.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp [hunk 2]
```diff
@@ -1380,6 +1408,34 @@ namespace Azure { namespace Storage { namespace Blobs {
...
+    /**
+     * @brief Optional conditions that must be met to perform this operation.
+     */
+    BlobAccessConditions AccessConditions;
+  };
+
+  /**
+   * @brief Optional parameters for PageBlobClient::GetManagedDiskPageRangesDiff.
+   */
+  struct GetManagedDiskPageRangesDiffOptions
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:1432]: Make `PreviousSnapshotUrl` required parameter in the corresponding API.

File: sdk/storage/azure-storage-blobs/src/page_blob_client.cpp [hunk 3]
```diff
@@ -233,8 +233,83 @@ namespace Azure { namespace Storage { namespace Blobs {
...
+    ret.LastModified = std::move(protocolLayerResponse->LastModified);
+    ret.BlobContentLength = protocolLayerResponse->BlobContentLength;
+    for (const auto& range : protocolLayerResponse->PageRanges)
+    {
+      ret.PageRanges.emplace_back(Models::PageRange{range.first, range.second - range.first + 1});
+    }
+    for (const auto& range : protocolLayerResponse->ClearRanges)
+    {
+      ret.ClearRanges.emplace_back(Models::PageRange{range.first, range.second - range.first + 1});
+    }
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/src/page_blob_client.cpp:299]: nit: Consider extracting the duplicated code to call the protocol layer to one place. A private helper function that takes `previousSnapshot`, `previousSnapshotUrl` and the option will do.

**Review by katmsft — APPROVED** [review]

File: sdk/storage/azure-storage-blobs/src/page_blob_client.cpp (hunk 3 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/src/page_blob_client.cpp:299]: I don't feel like doing that.

**Pull request merged**